### PR TITLE
Add a Shapes demo for DDS interoperability testing

### PR DESCRIPTION
**Pull request by jhelovuo**
Remake of pull request https://github.com/eclipse-zenoh/zenoh/pull/1310 in Zenoh main repo, as this is considered a more appropriate place.

### CODE REVIEW

File: zenoh-dds-interop/shapes_demo/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+# shapes-demo
+
+A program to test the interoperability of [zenoh-plugin-dds](https://github.com/eclipse-zenoh/zenoh-plugin-dds) against various DDS implementations.
+
+## Dependencies
+
+
+- [Rust](http://rust-lang.org) installation instructions are available [here](https://www.rust-lang.org/tools/install)
+- [zenoh-plugin-dds](https://github.com/eclipse-zenoh/zenoh-plugin-dds)
+- A DDS Shapes Demo application, using e.g. [RustDDS](https://atostek.com/en/services/rust-dds/) that has a built-in [Shapes Demo](https://github.com/jhelovuo/RustDDS/tree/master/examples/shapes_demo).
```
Comment (Mallets) [inline zenoh-dds-interop/shapes_demo/README.md:10]: It would be better to remove the link pointing to the commercial page of `RustDDS`.

Comment (JEnoch) [reply zenoh-dds-interop/shapes_demo/README.md:10]: Agreed. @jhelovuo could you please set https://github.com/jhelovuo/RustDDS as URL for `RustDDS`.
Also please add https://github.com/atolab/dds-ishapes as another DDS Shapes Demo using CycloneDDS.

Comment (JEnoch) [reply zenoh-dds-interop/shapes_demo/README.md]: Thanks you!

**Review by JEnoch — APPROVED** [review]
LGTM. Thank you for your contribution!

**Pull request merged**